modules/client/src/main/java/org/apache/ignite/jdbc/IgniteJdbcDriver.java

```diff
@@ -107,7 +107,7 @@
      * @return Driver instance.
      * @throws RuntimeException when failed to register driver.
      */
-    public static synchronized Driver register() {
+    private static synchronized Driver register() {
```

Comment:
```suggestion
    private static synchronized void register() {
```

Reply:
Let's drop 'registered' flag and make instance non-final.
See: https://github.com/pgjdbc/pgjdbc/blob/master/pgjdbc/src/main/java/org/postgresql/Driver.java